addons/mrp_account/models/product.py

```diff
@@ -55,15 +57,14 @@ def _compute_average_price(self, qty_invoiced, qty_to_invoice, stock_moves):
             return super()._compute_average_price(qty_invoiced, qty_to_invoice, stock_moves)
         value = 0
         dummy, bom_lines = bom.explode(self, 1)
-        bom_lines = {line: data for line, data in bom_lines}
-        for bom_line, moves_list in groupby(stock_moves.filtered(lambda sm: sm.state != 'cancel'), lambda sm: sm.bom_line_id):
-            if bom_line not in bom_lines:
-                for move in moves_list:
-                    value += move.product_qty * move.product_id._compute_average_price(qty_invoiced * move.product_qty, qty_to_invoice * move.product_qty, move)
-                continue
-            line_qty = bom_line.product_uom_id._compute_quantity(bom_lines[bom_line]['qty'], bom_line.product_id.uom_id)
+        product_qty_ratios = defaultdict(lambda: 0)
+        for line, data in bom_lines:
+            product_qty_ratios[line.product_id.id] += line.product_uom_id._compute_quantity(data['qty'], line.product_id.uom_id) / bom.product_qty
+
+        for product_id, moves_list in groupby(stock_moves.filtered(lambda sm: sm.state != 'cancel'), lambda sm: sm.product_id):
+            ratio = product_qty_ratios[product_id.id]
             moves = self.env['stock.move'].concat(*moves_list)
-            value += line_qty * bom_line.product_id._compute_average_price(qty_invoiced * line_qty, qty_to_invoice * line_qty, moves)
+            value += ratio * product_id._compute_average_price(qty_invoiced * ratio, qty_to_invoice * ratio, moves)
```

Comment:
doesn't `move.product_qty` missing in the computation ?

Reply:
using move.product_qty is actually wrong if the kit quantity was more than one all the components will be multiplied twice

Reply:
qty_to_invoice represents the required kit qty

Reply:
We should keep this specific case 
```python  
if bom_line not in bom_lines:
    for move in moves_list:
        value += move.product_qty * move.product_id._compute_average_price(qty_invoiced * move.product_qty, qty_to_invoice * move.product_qty, move)
        continue
```

that correspond to a change in the kit bom.
As different bom lines with the same product will create different stock moves in case of kit, we can keep the patch as it is with just a ` / bom.product_qty` in the value computation (both ` value += move.product_qty  ... ` and ` line_qty = ...`)

Reply:
if you have a Bom that uses 4 components and you try to sell 6 units of the Bom
move.product_qty will already be 24 and qty_to_invoice is 6 in this case multiplying by move.product_qty only works if we're selling 1 product